src/dcd/common/constants.d

```diff
@@ -26,87 +26,122 @@ struct ConstantCompletion
 	string ddoc;
 }
 
+private static immutable pragmaDDoc = import("pragma.dd");
+private static immutable traitsDDoc = import("traits.dd");
+
+template fetchDocDDforDT(string dtStart)
+{
+	enum fetchDocDDforDT = {
+		import std.string : lineSplitter, KeepTerminator, stripLeft, stripRight,
+			strip, startsWith, endsWith;
+
+		bool found;
+		string ret;
+		string indent;
+		bool inCode;
+		foreach (line; pragmaDDoc.lineSplitter!(KeepTerminator.yes))
+		{
+			if (line.stripLeft.startsWith("$(DT $(LNAME2 ")
+					|| line.stripLeft.startsWith("$(SPEC_SUBNAV_PREV_NEXT"))
+			{
+				if (found) // abort on new section if in section
+					break;
+				else if (line.stripLeft.startsWith("$(DT $(LNAME2 " ~ dtStart)) // set section if correct
+				{
+					indent = line[0 .. $ - line.stripLeft.length];
+					found = true;
+				}
+			}
+			else if (found)
+			{
+				if (line.startsWith("---")) // code blocks aren't indented
+					inCode = !inCode;
+
+				if (inCode)
+					ret ~= line;
+				else
+				{
+					if (line.startsWith(indent)) // strip indentation of DT
+						ret ~= line[indent.length .. $];
+					else
+						ret ~= line;
+				}
+			}
+		}
```

Comment:
This looks like an overkill. Why not sth. like

```d
alias needles = AliasSeq!("$(SECTION2 $(GNAME "~name~")", "$(H2 $(GNAME "~name~")");
auto searchText = ddocText.find(needles);
assert(!text.empty);
auto text = searchText[0].drop(searchText[1]);
// do any additional work with text
auto idx = text.representation.find("$(SECTION, "$(H2", "$(SPEC").walkLength;
return text[0 .. idx];
```

Reply:
I am also dropping some indentation at the start for non-code parts which is why I need to iterate over each line and do special checks